packages/graphiql/src/components/GraphiQL.js

```diff
@@ -120,6 +121,9 @@ export class GraphiQL extends React.Component {
       docExplorerOpen = this._storage.get('docExplorerOpen') === 'true';
     }
 
+    // initial variable editor pane open
+    const variableEditorOpen = props.defaultVariableEditorOpen !== undefined ? props.defaultVariableEditorOpen : Boolean(variables);
```

Comment:
(Note to self: comparing against `undefined` should be safe in modern browsers; I think this is okay.)